### PR TITLE
added `.zip_with` to MultiObject

### PR DESCRIPTION
**Pull request by koreno**

### CODE REVIEW

File: easypy/concurrency.py [hunk 1]
```diff
@@ -599,6 +599,10 @@ def chain(self):
         "Chain the iterables contained by this MultiObject"
         return self.__class__(chain(*self))
 
+    def zip_with(self, *collections):
+        assert all(len(coll) == len(self) for coll in collections), "All collection must be of length %s" % len(self)
```
Comment (idanarye) [inline easypy/concurrency.py]: Is this necessary? This makes it hard to use with infinite generators like `itertools.cycle`...

**Comment by idanarye** [comment]
Is the tokens commit supposed to be here?

**Comment by koreno** [comment]
Yeah, unrelated, but not worth It's own pr

File: easypy/concurrency.py (hunk 1 above)
Comment (koreno) [inline easypy/concurrency.py]: I think that's dangerous... at the very least `self.items` should all be paired, to avoid subtle mistakes

Comment (koreno) [inline easypy/concurrency.py]: It's the type of thing I could discard when I encounter an real use case

Comment (idanarye) [inline easypy/concurrency.py]: How about this then:

```python
def zip_with(self, *collections):
    result = self.__class__(zip(self, *collections))
    assert len(self) == len(result), "All collection must be of at least length %s" % len(self)
    return result
```

**Comment by koreno** [comment]
Clearly that's better
😒

**Pull request merged**